packages/batch-submitter/src/transaction-chain-contract.ts

```diff
@@ -19,6 +20,29 @@ export { encodeAppendSequencerBatch, BatchContext, AppendSequencerBatchParams }
  * where the `appendSequencerBatch(...)` function uses a specialized encoding for improved efficiency.
  */
 export class CanonicalTransactionChainContract extends Contract {
+  public customPopulateTransaction = {
+    appendSequencerBatch: async (
+      batch: AppendSequencerBatchParams,
+      options?: TransactionRequest
+    ): Promise<ethers.PopulatedTransaction> => {
+      const nonce = await this.signer.getTransactionCount()
+      const to = this.address
+      const data = getEncodedCalldata(batch)
+      const gasLimit = await this.signer.provider.estimateGas({
+        to,
+        from: await this.signer.getAddress(),
+        data,
+      })
```

Comment:
Isn't this the call that failed previously? Maybe I'm misunderstanding, but this is still doing a gas estimation without an override, so wouldn't this call still be including "pending" transactions as final in its gas estimation?